### PR TITLE
Migrate more-info configurator

### PR DESCRIPTION
**Pull request by bramkragten**
## Proposed change

Migrate more-info configurator to Lit

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [X] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/dialogs/more-info/controls/more-info-configurator.ts [hunk 1]
```diff
@@ -0,0 +1,128 @@
+import "@material/mwc-button";
+import { HassEntity } from "home-assistant-js-websocket";
+import { css, html, LitElement } from "lit";
+import { customElement, property, state } from "lit/decorators";
+import "../../../components/ha-alert";
+import "../../../components/ha-circular-progress";
+import "../../../components/ha-markdown";
+import "../../../components/ha-textfield";
+import { HomeAssistant } from "../../../types";
```
Comment (zsarnett) [inline src/dialogs/more-info/controls/more-info-configurator.ts:9]:
```suggestion
import type { HomeAssistant } from "../../../types";
```

File: src/dialogs/more-info/controls/more-info-configurator.ts [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+export class MoreInfoConfigurator extends LitElement {
+  @property({ attribute: false }) public hass!: HomeAssistant;
+
+  @property({ attribute: false }) public stateObj?: HassEntity;
+
+  @state() private _isConfiguring = false;
+
+  private _fieldInput = {};
+
+  render() {
```
Comment (zsarnett) [inline src/dialogs/more-info/controls/more-info-configurator.ts:21]:
```suggestion
  protected render() {
```

File: src/dialogs/more-info/controls/more-info-configurator.ts [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+        ${this.stateObj.attributes.fields.map(
+          (field) => html`<ha-textfield
+            .label=${field.name}
+            .name=${field.id}
+            .type=${field.type}
+            @change=${this._fieldChanged}
+          ></ha-textfield>`
+        )}
+        ${this.stateObj.attributes.submit_caption
+          ? html`<p class="submit">
```
Comment (zsarnett) [inline src/dialogs/more-info/controls/more-info-configurator.ts:47]: Does this need to be a paragraph element

**Review by zsarnett — DISMISSED** [review]

File: src/dialogs/more-info/controls/more-info-configurator.ts (hunk 3 above)
Comment (bramkragten) [reply src/dialogs/more-info/controls/more-info-configurator.ts:47]: No, could be a div, but 🤷‍♂️  (it is used to center the button)

**Pull request merged**